syz-verifier/main.go

```diff
@@ -298,6 +303,38 @@ func (vrf *Verifier) generate() (*prog.Prog, int) {
 	return vrf.target.Generate(vrf.rnd, prog.RecommendedCalls, vrf.choiceTable), vrf.progIdx
 }
 
+// processResults will send a set of complete results for verification and, in
+// case differences are found, it will store a result report highlighting those
+// in th workdir/results directory. If writing the results fails, it returns an
+// error.
+func (vrf *Verifier) processResults(res []*verf.Result, prog *prog.Prog) error {
+	rr := verf.Verify(res, prog)
+	if rr == nil {
+		log.Print("HEREEEEE")
+		return nil
+	}
+	data := fmt.Sprintf("Errno mismatches found for program:\n%s\n", rr.Prog)
+	data += "CALL REPORTS FOUND BELOW\n"
+
+	mismatchIdx := 0
+	for idx, cr := range rr.Reports {
+		data += fmt.Sprintf("Report for call: %s", cr.Call)
+
+		if mismatchIdx < len(rr.Mismatch) && idx == rr.Mismatch[mismatchIdx] {
```

Comment:
It seems this can be significantly simplified if you make Mismatch a field of CallResult.

Reply:
Yes, that is true, thank you!